app/shared/components/Contract/Interface/Form/Action.js

```diff
@@ -24,35 +26,43 @@ class ContractInterfaceFormAction extends Component<Props> {
       this.resetForm(nextProps.contractAction);
     }
   }
-  formatField = (contractAction, name, value = '') => {
+  formatField = (contractAction, name, value = '', type = null) => {
     const {
       contract
     } = this.props;
-    const fieldType = contract.getFieldType(contractAction, name);
+    let fieldType = contract.getFieldType(contractAction, name);
+
+    if (type === 'array') {
+      fieldType = 'array';
+    }
     switch (fieldType) {
       case 'int': {
         return parseInt(value, 10);
       }
       case 'bool': {
         return value ? 1 : 0;
       }
+      case 'array': {
+        return value;
+      }
       default: {
         return String(value);
       }
     }
-  }
-  onChange = (e, { name, value }) => {
+  };
+  onChange = (e, { name, value }, type = null) => {
     const { contractAction } = this.props;
+    console.log({ valueonchange: value });
```

Comment:
some debugging made it's way in, not planning to block the merge but just so you're aware :)